tutorials/SingleQubitGates/SingleQubitGates.ipynb

```diff
@@ -0,0 +1,874 @@
+    ">\n",
+    "> $$A|\\phi\\rangle = \\epsilon|\\phi\\rangle \\\\\n",
+    "A|\\psi\\rangle = \\zeta|\\psi\\rangle$$\n",
+    ">\n",
+    "> Then:\n",
+    ">\n",
+    "> $$A = \\epsilon|\\phi\\rangle\\langle\\phi| + \\zeta|\\psi\\rangle\\langle\\psi|$$\n",
+    ">\n",
+    "> Let's use our $X$ gate as a simple example. The $X$ gate has two eigenvectors: $|+\\rangle = \\frac{1}{\\sqrt{2}}\\big(|0\\rangle + |1\\rangle\\big)$ and $|-\\rangle = \\frac{1}{\\sqrt{2}}\\big(|0\\rangle - |1\\rangle\\big)$. Their eigenvalues are $1$ and $-1$ respectively:\n",
+    ">\n",
+    "> $$X|+\\rangle = |+\\rangle \\\\\n",
+    "X|-\\rangle = -|-\\rangle$$\n",
+    ">\n",
+    "> Here's what the decomposition looks like:\n",
+    ">\n",
+    "> $$X = |+\\rangle\\langle+| - |-\\rangle\\langle-| =\n",
+    "\\frac{1}{\\sqrt{2}}\\big(|0\\rangle + |1\\rangle\\big)\\frac{1}{\\sqrt{2}}\\big(\\langle0| + \\langle1|\\big) - |-\\rangle\\langle-| = \\\\\n",
+    "= \\frac{1}{2}\\big[\\big(|0\\rangle + |1\\rangle\\big)\\big(\\langle0| + \\langle1\\big) - \\big(|0\\rangle - |1\\rangle\\big)\\big(\\langle0| - \\langle1|\\big)\\big] = \\\\\n",
+    "= \\frac{1}{2}\\big(|0\\rangle\\langle0| + |0\\rangle\\langle1| + |1\\rangle\\langle0| + |1\\rangle\\langle1| - |0\\rangle\\langle0| + |0\\rangle\\langle1| + |1\\rangle\\langle0| - |1\\rangle\\langle1|\\big) = \\\\\n",
+    "= \\frac{1}{2}\\big(|0\\rangle\\langle0| - |0\\rangle\\langle0| + |0\\rangle\\langle1| + |0\\rangle\\langle1| + |1\\rangle\\langle0| + |1\\rangle\\langle0| + |1\\rangle\\langle1| - |1\\rangle\\langle1|\\big) =\n",
+    "\\frac{1}{2}\\big(2|0\\rangle\\langle1| + 2|1\\rangle\\langle0|\\big) = |0\\rangle\\langle1| + |1\\rangle\\langle0|$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Important Gates\n",
+    "\n",
+    "This section introduces some of the common single qubit gates, including their matrix form, their ket-bra decomposition, and a brief summary of what they do to various qubit states.\n",
+    "\n",
+    "This section relies on the following notation:\n",
+    "\n",
+    "$$|+\\rangle = \\frac{1}{\\sqrt{2}}\\big(|0\\rangle + |1\\rangle\\big) \\\\\n",
+    "|-\\rangle = \\frac{1}{\\sqrt{2}}\\big(|0\\rangle - |1\\rangle\\big) \\\\\n",
+    "|i\\rangle = \\frac{1}{\\sqrt{2}}\\big(|0\\rangle + i|1\\rangle\\big) \\\\\n",
+    "|-i\\rangle = \\frac{1}{\\sqrt{2}}\\big(|0\\rangle - i|1\\rangle\\big) \\\\\n",
+    "|\\psi\\rangle = \\alpha|0\\rangle + \\beta|1\\rangle$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Pauli Gates\n",
+    "\n",
+    "The Pauli gates, named after Wolfgang Pauli ([Wikipedia](https://en.wikipedia.org/wiki/Wolfgang_Pauli)), are based on the so-called **Pauli matrices**, which also have some special significance to the topic of measurement. All three Pauli gates are **self-adjoint**, meaning that each one is its own inverse.\n",
+    "\n",
+    "### Pauli $X$ Gate\n",
```

Comment:
I wouldn't add a full section for each gate before the table, especially since the text in the sections is rather short. And X being the most common gate is arguable - H can give it a run for its money. Maybe a notes section after the table telling that X gate is equivalent of classical NOT and is known as bit flip and Z gate is known as phase flip would be sufficient? I'm worried that this tutorial has grown rather long and intimidating...